src/Microsoft.OData.ModelBuilder/Vocabularies/Capabilities/DeleteRestrictionsBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.OData.ModelBuilder.Vocabularies.Capabilities
+{
+    /// <summary>
+    /// Allows building restrictions on delete operations;
+    /// </summary>
+    public class DeleteRestrictionsBuilder : VocabularyBuilder<DeleteRestrictionsType>
+    {
+        /// <summary>
+        /// Initializes a new <see cref="DeleteRestrictionsBuilder"/>
+        /// </summary>
+        public DeleteRestrictionsBuilder()
+            : base("Org.OData.Capabilities.V1.DeleteRestrictions", new DeleteRestrictionsType())
+        {
+        }
+
+        /// <summary>
+        /// Entities can be deleted.
+        /// </summary>
+        /// <param name="deletable">Entities can be deleted.</param>
+        /// <returns><see cref="DeleteRestrictionsBuilder" /></returns>
+        public DeleteRestrictionsBuilder IsDeletable(bool deletable)
+        {
+            Data.Deletable = deletable;
+            return this;
+        }
+
+        /// <summary>
+        /// Required permissions. One of the specified sets of scopes is required to perform the delete.
+        /// </summary>
+        /// <param name="permissions">Required permissions. One of the specified sets of scopes is required to perform the delete.</param>
+        /// <returns><see cref="DeleteRestrictionsBuilder" /></returns>
+        public DeleteRestrictionsBuilder HasPermissions(IEnumerable<PermissionType> permissions)
```

Comment:
Using 'Has' confuses the reader since one expects that if the object either has or doesn't have the property  the method will return true/false

Reply:
Well... this follows fluent api conventions already established in entity framework from even before efcore as can be seen in the ef6 documentation on [fluent api](https://docs.microsoft.com/en-us/ef/ef6/modeling/code-first/fluent/types-and-properties)